### PR TITLE
CDPSDX-3936 add hive direct sql configatrion to the blueprints

### PR DESCRIPTION
**Pull request by RokolyA94**
CDPSDX-3936 add hive direct sql configatrion to the blueprints

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveKnoxConfigProvider.java [hunk 1]
```diff
@@ -25,15 +27,20 @@ public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor temp
...
             String realm = kerberosConfigOpt.getRealm();
             String principal = ConfigUtils.getSafetyValveProperty("hive.server2.authentication.spnego.principal", "HTTP/_HOST@" + realm);
             String keytab = ConfigUtils.getSafetyValveProperty("hive.server2.authentication.spnego.keytab", "hive.keytab");
-            return List.of(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, principal + keytab + filePerEvent));
+            configs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, principal + keytab + filePerEvent));
         }
+        if(isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERA_STACK_VERSION_7_2_10)){
+            configs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, "<property><name>hive.metastore.try.direct.sql.ddl</name><value>true</value></property><property><name>hive.metastore.try.direct.sql</name><value>true</value></property>"));
+        }
+        return configs;
```
Comment (kkalvagadda1) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveKnoxConfigProvider.java:43]: Why is this change needed for changing the HMS configuration for Md Ha blueprints with profilers?

**Comment by doktoric** [comment]
checkstyle failed

**Review by romedeiro — APPROVED** [review]
Small thing, typo in the commit message "configatrion"

**Review by sxxgrc — APPROVED** [review]

**Review by grace303303 — APPROVED** [review]

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProvider.java
Comment (lajosrodek) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProvider.java:1]: Please cover with UT.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProvider.java [hunk 2]
```diff
@@ -101,8 +101,12 @@ public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor temp
 
         Optional<KerberosConfig> kerberosConfigOpt = source.getKerberosConfig();
         if (kerberosConfigOpt.isPresent()) {
-            String safetyValveValue = getSafetyValveProperty("hive.hook.proto.file.per.event", Boolean.TRUE.toString());
-            configs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, safetyValveValue));
+            StringBuilder safetyValveValue = new StringBuilder(getSafetyValveProperty("hive.hook.proto.file.per.event", Boolean.TRUE.toString()));
+            if (source.getStackType() != null && source.getStackType().equals(StackType.DATALAKE)) {
+                safetyValveValue.append("<property><name>hive.metastore.try.direct.sql.ddl</name><value>true</value></property>");
+                safetyValveValue.append("<property><name>hive.metastore.try.direct.sql</name><value>true</value></property>");
+            }
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProvider.java:108]: What is the rationale behind moving this out from the bp and adding it dynamically?

**Review by lajosrodek — CHANGES_REQUESTED** [review]

**Comment by lajosrodek** [comment]
@RokolyA94 

CheckStyle error:
```
> Task :template-manager-cmtemplate:checkstyleTest
[ant:checkstyle] [ERROR] /grid/0/jenkins/workspace/workspace/cloudbreak-pull-request-builder-checkstyle-test/template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProviderTest.java:368: Line is longer than 161 characters (found 167). [LineLength]
```

plus there 16 UT failures; see [here[(http://build.eng.hortonworks.com/job/cloudbreak-pull-request-builder-unit-test/638/).

**Review by horadla23 — APPROVED** [review]

**Pull request merged**